playbooks/provisioning/openstack/provision-openstack.yml

```diff
@@ -31,12 +32,13 @@
 - name: Refresh Server inventory
   hosts: localhost
   connection: local
+  become: False
   gather_facts: False
   tasks:
   - meta: refresh_inventory
 
 - hosts: cluster_hosts
-  gather_facts: false
+  gather_facts: True
```

Comment:
note: this is a trick to collect all facts for cluster nodes while waiting for provisioned servers to come up. I hope that works.